sdk/enter.go

```diff
@@ -80,6 +80,62 @@ func simpleChrootHelper(args []string) error {
 		return fmt.Errorf("Mounting %q failed: %v", newRepoRoot, err)
 	}
 
+	homedir := os.Getenv("HOME")
+	if homedir == "" {
+		return fmt.Errorf("HOME environment variable is not set.")
+	}
+
+	// mount the directory containing the ssh-agent socket in order
+	// to support a private repos during repo sync
+	sshAuthSock := os.Getenv("SSH_AUTH_SOCK")
+	if sshAuthSock != "" {
+		sshSourceDir := filepath.Dir(sshAuthSock)
+		if _, err := os.Stat(sshSourceDir); err == nil {
+			sshTargetdir := filepath.Join(chroot, sshSourceDir)
+
+			if err := os.MkdirAll(sshTargetdir, 0700); err != nil {
+				return err
+			}
+
+			if err := syscall.Mount(
+				sshSourceDir, sshTargetdir, "none", syscall.MS_BIND, ""); err != nil {
+				return fmt.Errorf("Mounting %q failed: %v", sshSourceDir, err)
+			}
+		}
+
+		if err := os.MkdirAll(filepath.Join(chroot, ".ssh"), 0700); err != nil {
```

Comment:
Everything from here down doesn't need to run as root and in fact probably shouldn't so the copied files get the right ownership. Also it should be perfectly fine to copy ssh configs even if there isn't an agent socket, iirc that's what the usual enter_chroot.sh script does. So the rest of this can get move to another function that gets called before `sudo.Run()` in `SimpleEnter`. Also, given the common boiler plate for all the copies a helper `copyToChroot` would help clean that up.